src/com/facebook/buck/android/MergeAndroidResourcesStep.java

```diff
@@ -339,7 +339,7 @@ void writePerPackageRDotJava(
     SortedSetMultimap<RDotTxtEntry, Path> bannedDuplicateResourceToSymbolsFiles =
         TreeMultimap.create();
 
-    HashMap<RDotTxtEntry, String> resourceToIdValuesMap = new HashMap<>();
+    HashMap<Integer, RDotTxtEntry> resourceToIdValuesMap = new HashMap<>();
```

Comment:
why do we need `HashMap<Integer, RDotTxtEntry>`? we are only using the idValue below. 

I think it should atleast be `HashMap<RDotTxtEntry, RDotTxtEntry>` so that `equals()` on `RDotTxtEntry` is also called.